### PR TITLE
[TF 2.0 API Docs] tf.identity

### PR DESCRIPTION
**Pull request by imransalam**
Added usage example in the identity function and and added a value error raise for None input provided in the doc string and also in the code.
The related issue is 
https://github.com/tensorflow/tensorflow/issues/29315

### CODE REVIEW

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

File: tensorflow/python/ops/array_ops.py [hunk 1]
```diff
@@ -67,7 +67,24 @@ def identity(input, name=None):  # pylint: disable=redefined-builtin
...
   Returns:
     A `Tensor`. Has the same type as `input`.
+    
+  Raises:
+    ValueError: If input is None.
+    
+  Usage Example:
+    ```python
+    >> import tensorflow as tf
+    >> x = tf.zeros(shape=(3,3)) + 4
```
Comment (steph-en-m) [inline tensorflow/python/ops/array_ops.py]: * You don't need the ```import tensorflow as tf``` statement and also  (```python ...) so omit them.
* doctests use 3 `>` signs not two as your using here.
Here is a guide to help you through https://www.tensorflow.org/community/contribute/docs_ref

**Review by stmugisha — CHANGES_REQUESTED** [review]

File: tensorflow/python/ops/array_ops.py (hunk 1 above)
Comment (steph-en-m) [reply tensorflow/python/ops/array_ops.py]: Also first ensure your fork is up-to date with the tensorflow main repo before performing any further changes as there seems to be some conflicting files.

Comment (imransalam) [reply tensorflow/python/ops/array_ops.py]: Done

File: tensorflow/python/ops/array_ops.py [hunk 2]
```diff
@@ -264,9 +264,23 @@ def identity(input, name=None):  # pylint: disable=redefined-builtin
 
   Returns:
     A `Tensor`. Has the same type as `input`.
+    
+  Raises:
+    ValueError: If input is None.
+    
+  Usage Example:
+    >>> x = tf.zeros(shape=(3,3)) + 4
```
Comment (steph-en-m) [inline tensorflow/python/ops/array_ops.py]: This ```tf.zeros(shape=(3,3)) + 4``` looks a bit mechanical and can be done more neatly using
```tf.constant(4, shape=(3,3))```

File: tensorflow/python/ops/array_ops.py [hunk 3]
```diff
@@ -264,9 +264,23 @@ def identity(input, name=None):  # pylint: disable=redefined-builtin
...
+    >>> y = tf.identity(x)
+    >>> print(y)
```
Comment (steph-en-m) [inline tensorflow/python/ops/array_ops.py]: These two statements can be reduced into one statement ```>>> tf.identity(x)``` to avoid redundancy.

**Review by stmugisha — CHANGES_REQUESTED** [review]

File: tensorflow/python/ops/array_ops.py [hunk 4]
```diff
@@ -269,13 +269,12 @@ def identity(input, name=None):  # pylint: disable=redefined-builtin
...
-    tf.Tensor(
-    [[4. 4. 4.]
-     [4. 4. 4.]
-     [4. 4. 4.]], shape=(3, 3), dtype=float32)
+    >>> x = tf.constant(4, shape=(3,3))
+    >>> tf.identity(x)
+    <tf.Tensor: id=2, shape=(3, 3), dtype=int32, numpy=
+    array([[4, 4, 4],
+       [4, 4, 4],
+       [4, 4, 4]], dtype=int32)>
```
Comment (steph-en-m) [inline tensorflow/python/ops/array_ops.py]: Indent the two rows below `array([[4, 4, 4],`  to the same level as the first row (`[4,4,4]`)

**Review by stmugisha — CHANGES_REQUESTED** [review]

File: tensorflow/python/ops/array_ops.py [hunk 5]
```diff
@@ -67,7 +67,24 @@ def identity(input, name=None):  # pylint: disable=redefined-builtin
...
+    >> y = tf.identity(x)
+    >> print(y)
+    tf.Tensor(
+    [[4. 4. 4.]
+     [4. 4. 4.]
+     [4. 4. 4.]], shape=(3, 3), dtype=float32)
+     ```
   """
+  if input is None:
+    raise ValueError("The input Specified is None. Must specified an input to tf.identity()")
```
Comment (steph-en-m) [inline tensorflow/python/ops/array_ops.py]: I think this exception is captured under 
```
if context.executing_eagerly() and not hasattr(input, "graph"):
    input = ops.convert_to_tensor(input)
    in_device = input.backing_device 
```
when an attempt to convert a none input to a tensor fails (`ValueError`). So I guess this additional `if input is None` isn't needed.

**Comment by steph-en-m** [comment]
@rthadur I've realised that there already exists a usage example for `tf.identity` (but not in the doctest format). So  should we take this as an update or leave it?

File: tensorflow/python/ops/array_ops.py [hunk 6]
```diff
@@ -67,7 +67,24 @@ def identity(input, name=None):  # pylint: disable=redefined-builtin
...
+    >> x = tf.zeros(shape=(3,3)) + 4
+    >> y = tf.identity(x)
+    >> print(y)
+    tf.Tensor(
+    [[4. 4. 4.]
+     [4. 4. 4.]
+     [4. 4. 4.]], shape=(3, 3), dtype=float32)
+     ```
   """
+  if input is None:
```
Comment (steph-en-m) [inline tensorflow/python/ops/array_ops.py]: This exception is captured under 
```
if context.executing_eagerly() and not hasattr(input, "graph"):
    input = ops.convert_to_tensor(input)
    in_device = input.backing_device 
```
as an attempt to convert a `None` input to a tensor raises a `ValueError`.
So I guess this additional `if input is None` isn't needed.

**Review by stmugisha — CHANGES_REQUESTED** [review]

**Comment by steph-en-m** [comment]
@rthadur I've realised that there already exists a usage example for tf.identity (but not in the doctest format). So should this be taken as an update or leave it?

**Review by stmugisha — COMMENTED** [review]
place your usage example in the section below the existing one.

**Comment by gbaned** [comment]
@imransalam Can you please resolve conflicts? Thanks!

**Comment by gbaned** [comment]
@imransalam thank you, it is failing doctest can you please check here for [logs](https://source.cloud.google.com/results/invocations/a50cfffd-1e55-4b87-a2ad-9079d680c470/targets/%2F%2Ftensorflow%2Ftools%2Fdocs:tf_doctest/tests).

Please run the doctest locally as mentioned here in the [contributor guidelines](https://www.tensorflow.org/community/contribute/docs_ref).

**Comment by tensorflowbutler** [comment]
It has been 15 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by gbaned** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!

**Pull request closed without merging**